### DISCUSSION TITLE
Trying to document custom response classes

### DISCUSSION THREAD

**Discussion by Wulfheart** [dedoc/scramble#179] [Q&A]
Hello,

I am currently trying to document custom response classes.
For the response data I have the following class:
```php
final class ResponseData
{
    public function __construct(
        public NestedId $id,
    )
    {
    }

    public function toArray(): array
    {
        return [
            /** @var NestedId $id */
            'nestedId' => $this->id,
            /** @var Channel $channel */
            'channel' => ChannelDataProvider::getRandomEntity(),
        ];
    }
}
```

I return `new JsonResponse($data->toArray())` in my controller. 
`NestedId` itself looks like this:
```php
readonly class Nested implements \JsonSerializable
{
    public function __construct(
        public string $id
    ) {
    }
    
   public function jsonSerialize(): mixed
   {
       return $this->id;
   }
    
}
```
I wrote a custom extension which adds the uuid format to the `nestedId` property. However, I would like to get rid of the `/** @var NestedId $id */`. When I do this it says that it is only of type `string` in the OpenAPI and nothing about the format. What steps do I have to take to have this still? Thank you in advance.

**Comment by Wulfheart** [comment]
Update: As far as I can see it is also impossible to do this via an extension. Maybe @romalytvynenko knows more about this. I am definitely out of ideas.

**Reply by romalytvynenko** [reply]
@Wulfheart please share your extension code as well, I'll take a look when I'd work on Scramble tasks

**Reply by Wulfheart** [reply]
I don't have it right now as I tinkered around with multiple approaches. My problem is that I have the types as `UnknownType` in the `OperationExtension`.

**Reply by romalytvynenko** [reply]
To be honest, I think the issue here is that Scramble (most likely) doesn't work with constructor property promotion well. Not sure about this, but I will check. I feel like the code you share should be analyzable pretty well.

**Reply by Wulfheart** [reply]
I have to try this out if it works  with non-promoted properties.

**Reply by Wulfheart** [reply]
Nope, no luck.
